From 3e21cd632b9a1e0b5c563cb2bb1c7ecbcbddc637 Mon Sep 17 00:00:00 2001
From: Brian Bennett <brian.bennett@joyent.com>
Date: Wed, 2 May 2018 09:49:09 -0700
Subject: [PATCH] TRITON-335 Better "latest" links for SmartDataCenter and
 platform Reviewed by: Trent Mick <trentm@gmail.com> Approved by: Trent Mick
 <trentm@gmail.com>

---
 tools/mantaput-bits | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/mantaput-bits b/tools/mantaput-bits
index 23043c0..7acc80f 100755
--- a/tools/mantaput-bits
+++ b/tools/mantaput-bits
@@ -122,9 +122,13 @@ done
 # upload the md5sums and config.mk
 echo -e $md5sums | mput -v -H 'content-type: text/plain' /${MANTA_USER}${UPLOAD_BASE_DIR}/${upload_subdir}/md5sums.txt
 mput -v -f $BITS_DIR/config.mk /${MANTA_USER}${UPLOAD_BASE_DIR}/${upload_subdir}/config.mk
-# now update the latest link
 
+# now update the branch latest link
 echo "/${MANTA_USER}${UPLOAD_BASE_DIR}/${upload_subdir}" | mput -v -H 'content-type: text/plain' /${MANTA_USER}${UPLOAD_BASE_DIR}/${UPLOAD_BRANCH}-latest
+# If this is a bi-weekly release branch, also update latest-release link
+if [[ $UPLOAD_BRANCH =~ ^release- ]]; then
+    echo "/${MANTA_USER}${UPLOAD_BASE_DIR}/${upload_subdir}" | mput -v -H 'content-type: text/plain' /${MANTA_USER}${UPLOAD_BASE_DIR}/latest-release
+fi
 
 echo "Uploaded to /${MANTA_USER}${UPLOAD_BASE_DIR}/${upload_subdir}"
 
-- 
2.21.0

